railties/test/generators/app_generator_test.rb

```diff
@@ -1042,6 +1042,35 @@ def test_version_control_initializes_git_repo
     assert_directory ".git"
   end
 
+  def test_default_branch_main_without_user_default
+    current_default_branch = `git config --global init.defaultBranch`
+    `git config --global --unset init.defaultBranch`
+
+    run_generator [destination_root]
+    assert_file ".git/HEAD", /main/
+
+    if !current_default_branch.strip.empty?
+      `git config --global init.defaultBranch #{current_default_branch}`
+    end
+  end
+
+  def test_version_control_initializes_git_repo_with_user_default_branch
+    git_version = `git --version`[/\d+.\d+.\d+/]
+    return if Gem::Version.new(git_version) < Gem::Version.new("2.28.0")
+
+    current_default_branch = `git config --global init.defaultBranch`
+    `git config --global init.defaultBranch master`
+
+    run_generator [destination_root]
+    assert_file ".git/HEAD", /master/
+
+    if current_default_branch.strip.empty?
+      `git config --global --unset init.defaultBranch`
+    else
+      `git config --global init.defaultBranch #{current_default_branch}`
+    end
```

Comment:
also something that should be inside an ensure block. otherwise a failing test would permanently add a default branch to a suite-runners's git config